### ISSUE TITLE
Eventing-RabbitMQ sets high CPU requests for dispatchers and is not configurable

### ISSUE THREAD

**Issue by lenalebt** [knative-sandbox/eventing-rabbitmq#813]
Hi folks,
was unsure whether to classify this as bug or feature requests. Sits somewhere in the middle imho...

**Describe the bug**
I just updated to eventing-rabbitMQ 1.5 and needed to downgrade again to 1.4 because suddenly the dispatcher instances request 300mCPU, which is way too much for our dev cluster (and prod as well).

 **Expected behavior**
CPU requests are configurable.

 **To Reproduce**
Install with default configuration.

 **Knative release version**
1.5

 **Additional context**
I do understand the values come from a load test, and also do understand that those may be good defaults, but in the end we can't afford to pay that much for development instances that don't really need those requests. It would be really nice to be able to just set a low request and live with the performance impact then.
I'd really love to update to 1.5 as well because our default-broker looses connections to rabbitmq pretty reliably (we're running DEV on preemptible nodes and hit the problem about once a day ;) ) and there have been fixes for those issues on 1.5. It's a kind of a dilemma - upgrading is problematic for other reasons than not upgrading :D.

Cheers, and keep up the good work :-)!

**Comment by gabo1208** [comment]
Hey @lenalebt ! Thanks for reaching out!

Very clear report, thanks for that. This is a use case we definitely missed while approximating resource requests so for the 1.5 release we will issue a patch (I think the job that creates the patch releases runs tomorrow, so tomorrow the v1.5.1 will be ready to go).

Regarding exposing this values to the user, this could need a little bit more discussion, so I'll close this issue as soon as the patch is ready an open other one to further discuss that point.

Thank you and so do you =)!

**Comment by lenalebt** [comment]
Thanks for the update and quick response :smile: !

I'll give a bit of context around our system why I think it makes sense to allow users to set the values somehow. I can repost it to another issue later on if needed.

We're having a few `Trigger` definitions per deployed application, but for some the events only occur rarely (say, a few events/minute). For others, they occur often (say, a few 10/second) - all not too fancy. I would love to only have one dispatcher per subscriber ref over all triggers (making it consume multiple RabbitMQ queues), but I do understand that this probably makes the implementation more complicated. If we had only one dispatcher, resource requests would be easier to accept :-). For the dispatchers with more load I'm happy to accept more resources, for the ones basically idling for quite some time - not so much...

Do you support the new trigger filters (https://knative.dev/docs/eventing/experimental-features/new-trigger-filters/) already? They look like it would allow to get less dispatchers per application, but I may be wrong there. In that case I could maybe just refactor the current `Trigger` definitions.

**Comment by gabo1208** [comment]
We still don't support the new trigger filters, we are planning to get to the MVP next release and start implementing more advanced features like that afterwards.

But we do support simple filters, that may help also, an example of this is located here:
[Multi consumer Broker-Trigger perf test](https://github.com/knative-sandbox/eventing-rabbitmq/blob/main/test/performance/broker-setup/400-broker-multi-consumer-setup.yaml)

Let me know if that helps :)

**Comment by gabo1208** [comment]
And yep @lenalebt if you can fill a feature request with this =)

**Comment by lenalebt** [comment]
Hey!

> But we do support simple filters, that may help also, an example of this is located here:
The simple filters don't really help as they create a dispatcher pod per Trigger. I have about 10 events for each of my deployments, each with a low event volume, making it 10 Triggers currently per deployment. I can now either start filtering in my service and remove the simple filter (and hammer it with all the other events that have higher volumes sometimes, adding load to the system where I don't need it). Or I can require loads of CPU that is really not needed for those cases. Both don't sound that appealing to me when I could just set a really low CPU request and be done (although it still eats RAM unnecessarily). CPU really is not an issue on those dispatchers.

I'm a bit sceptical about the KEDA autoscaler. My experience with Knative Serving scale-to-zero was not particularly positive, to the extent that I moved all my services back to simple k8s deployments because I had availability issues with services. I think the activator pods went crazy from time to time, and I did not get it under control to the extent that I'd trust it enough - but it's some months ago already. I'm only on Eventing currently, and completely moved away from Serving for that reason. I could not really point to the thing that went wrong, but removing that part of complexity made it work well.

I'll create a feature request for manually setting CPU requests and link it here.
